2020/dec05/dec05.go

```diff
@@ -13,5 +18,34 @@ type dec05 struct {
 }
 
 func (d dec05) Solve() (string, error) {
-	return "", nil
+	// we need to keep the highest and lowest seats
+	var highest BoardingPass
+	lowest := BoardingPass{math.MaxInt16, math.MaxInt16} // int16 should leave ample room
+	// we also need the sum of all seat IDs and the total number of boarding passes we scanned
+	var sum int64 // let's be generous
+	var length int
+	err := fileread.ReadAndApply(d.input, func(s string) error {
+		bp, err := NewBoardingPass(strings.Trim(s, " \t"))
+		if err != nil {
+			return err
+		}
+		if bp.IsHigher(highest) {
+			highest = bp
+		}
+		if lowest.IsHigher(bp) {
+			lowest = bp
+		}
+		sum += int64(bp.SeatID())
+		length++
+		return nil
+	})
+	if err != nil {
+		return "", err
+	}
+	length++
+	// there is probably a problem of luck depending on whether the missing seat is after or before the middle of the plane
+	// we compute the expected sum, were all seats actually figuring in the boarding passes
+	expectedSum := int64((highest.SeatID()+lowest.SeatID())*length) / 2
+	// then we substract the actual sum and return the difference
+	return fmt.Sprintf("The highest seat ID is %d", expectedSum-sum), nil
```

Comment:
that was smart :)

Reply:
Not my idea. I'm bad at inventing algorithms, and rather good at googling them.